### DISCUSSION TITLE
Reducing gas consumption

### DISCUSSION THREAD

**Discussion by Ortiz2525** [Ortiz2525/yield-farming#5] [Q&A] [answered]
How can get gas report of each function of contracts in hardhat?

**Comment by bony000** [comment, marked as answer]
First, you have to install hardhat-gas-reporter by this command
`npm install hardhat-gas-reporter --save-dev`

Then import this module in hardhat-config.ts
`import "hardhat-gas-reporter";`

Then add this to your config object
`gasReporter: {
    currency: 'CHF',
    gasPrice: 21,
    outputFile: 'gas-report.txt',
    noColors: true,
    onlyCalledMethods: true
  }`
